editor/editor_file_system.cpp

```diff
@@ -2773,6 +2773,34 @@ void EditorFileSystem::reimport_file_with_custom_parameters(const String &p_file
 	emit_signal(SNAME("resources_reimported"), reloads);
 }
 
+void EditorFileSystem::_queue_refresh_filesystem() {
+	if (refresh_queued) {
+		return;
+	}
+	refresh_queued = true;
+	get_tree()->connect(SNAME("process_frame"), callable_mp(this, &EditorFileSystem::_refresh_filesystem), CONNECT_ONE_SHOT);
+}
+
+void EditorFileSystem::_refresh_filesystem() {
+	for (EditorFileSystemDirectory *dir : folders_to_refresh) {
+		dir->subdirs.sort_custom<DirectoryComparator>();
+	}
+	folders_to_refresh.clear();
+
+	if (!files_to_refresh.is_empty()) {
```

Comment:
Do you think it would be possible to use `_update_scan_actions` instead of doing it yourself here?

1. Right now the update_files is done before the reimport_files. In `_update_scan_actions`, it's the other way around and I think there's a reason for that, it's to be sure that all the resources are imported when the scripts will be loaded. For example, to prevents tools scripts in addons to break when they use preload resources or to have the correct icon for scripts.

2. `_scan_extensions` is not executed. OK, I guess nobody should duplicate a folder with a .gdextension in it!!

3. Also, some duplications are done like the ACTION_DIR_ADD that moves the folder at the right place

I'm not saying it should absolutely use `_update_scan_actions`, it's more a thought that could prevent code and behavior duplication.

Reply:
Using `_update_scan_actions()` I'd have to manually add actions to `scan_actions`, which either requires refactoring how they are created or duplicating code. `_refresh_filesystem()` is a very small method, it doesn't really duplicate anything.

Reply:
Maybe call `_scan_fs_changes` after duplicating the directory. It can takes a `EditorFileSystemDirectory` and scan only this folder and subfolder?!?! Just an idea!